app/controllers/users_controller.rb

```diff
@@ -1,19 +1,18 @@
 class UsersController < ApplicationController
 
   def index
-
+    @users = User.all
   end
 
   def show
-
-  end
-
-  def new
-
+    @users = User.find(params[:id])
+    redirect_to(users_url)
   end
 
-  def destroy
+  private
```

Comment:
in controllers it is better use protected rather than private because of inheritance reasons. As private will not be inherited if you do a child controller from first one. Please replace with protected